ldap/servers/slapd/conntable.c

```diff
@@ -461,14 +449,10 @@ connection_table_move_connection_out_of_active_list(Connection_Table *ct, Connec
 
     /* Finally, place the connection back into the freelist for use */
     PR_ASSERT(c->c_refcnt == 0);
-    PR_ASSERT(ct->conn_free_offset != 0);
-    PR_ASSERT(ct->c_freelist[ct->conn_free_offset] == NULL);
-    ct->c_freelist[ct->conn_free_offset] = c;
-    ct->conn_free_offset = (ct->conn_free_offset + 1) % ct->size;
-    if (ct->conn_free_offset == 0) {
-        /* Never use slot 0 */
-        ct->conn_free_offset += 1;
-    }
+    PR_ASSERT(ct->conn_next_offset != 0);
```

Comment:
Could you confirm ? I understood that the range [0..ct_size-1] are valid values in the freelist.

Reply:
The PR_ASSERT is correct.
AFAIK the free list range is [0..ct_size].
But the various PR_ASSERT checks some specific conditions depending of the assert location.
In connection_table_move_connection_out_of_active_list code,  the connection that is moved should still be open (even if it may be in the closing state) so the free list cannot be full (i.e ct->conn_next_offset > 0)